File: android-exercises/3 - intent/Intent_java/app/src/main/java/com/bugrahankaramollaoglu/intent_java/MainActivity.java
package com.bugrahankaramollaoglu.intent_java;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    private EditText nameText;
    private EditText ageText;
    private Button changeButton;
    private String name;
    private String age;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        changeButton = (Button) findViewById(R.id.button);
        nameText = findViewById(R.id.nameText);
        ageText = findViewById(R.id.ageText);

        changeButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (nameText.getText().toString().isEmpty() || ageText.getText().toString().isEmpty()) {
                    Toast.makeText(MainActivity.this, "enter both", Toast.LENGTH_SHORT).show();
                } else {
                    name = nameText.getText().toString();
                    age = ageText.getText().toString();
                    Intent intent = new Intent(MainActivity.this, MainActivity2.class);
                    intent.putExtra("namekey", name);
                    intent.putExtra("agekey", age);
                    startActivity(intent);
                    finish(); // dersen önceki aktiviteleri kapatır, geri tuşuyla gidemezsin
                }
            }
        });
    }


/*  or you can set onClick as change and use this instead of setOnClickListener

    public void change(View view) {
        Intent intent = new Intent(this, MainActivity2.class);
        startActivity(intent);
    }   */
}

File: android-exercises/3 - intent/Intent_kotlin/app/src/main/java/com/bugrahankaramollaoglu/intent_kotlin/MainActivity.kt
package com.bugrahankaramollaoglu.intent_kotlin

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    private lateinit var button: Button
    private lateinit var nameText: EditText
    private lateinit var ageText: EditText
    private lateinit var name: String
    private lateinit var age: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button = findViewById<Button>(R.id.button)
        nameText = findViewById<EditText>(R.id.nameTextView)
        ageText = findViewById<EditText>(R.id.ageTextView)

        button.setOnClickListener {
            if (nameText.text.toString().isEmpty() || ageText.text.toString().isEmpty()) {
                Toast.makeText(this@MainActivity, "they cannot be empty", Toast.LENGTH_SHORT).show()
            } else {
                name = nameText.text.toString()
                age = ageText.text.toString()

                val intent: Intent = Intent(this@MainActivity, MainActivity2::class.java)
                intent.putExtra("namekey", name)
                intent.putExtra("agekey", age)
                startActivity(intent)
            }
        }
    }
}
